compiler/noirc_frontend/src/hir/comptime/interpreter.rs

```diff
@@ -219,7 +249,7 @@ impl<'a> Interpreter<'a> {
         argument: Value,
         location: Location,
     ) -> IResult<()> {
-        self.type_check(typ, &argument, location)?;
+        // self.type_check(typ, &argument, location)?;
```

Comment:
why is this disabled?

Reply:
This is failing for generic types since it sees `i32 != T` or similar.
We'll need another future PR to use instantiation bindings like the monomorphizer does. I've added a comment of this.

Reply:
There is https://github.com/noir-lang/noir/issues/4924 for getting generic function calls working.
This is temporary while they're still not working in the general case.

Reply:
This seems to have broken compilation of `noir-protocol-circuits` in as we end up attempting to unwrap a `None` https://github.com/AztecProtocol/aztec-packages/pull/6986